### ISSUE TITLE
Infinite recursion in baseURL

### ISSUE THREAD

**Issue by Duncan00** [softonic/axios-retry#67]
This issue is still exists:
https://github.com/softonic/axios-retry/issues/16

We created a client, with retry.
```
	const apiClient = axios.create({
		baseURL,
		timeout: 30000, // 30s
		headers: Object.assign(
			{
				'Content-Type': 'application/json',
			},
			headers
		),
		httpAgent,
		httpsAgent
	});
```

When retry, the baseURL will keep adding to prefix.

**Comment by ivoiv** [comment]
I think you're doing something wrong elsewhere because it works fine for us.
Remember baseURL should be an absolute URL, not a relative.

`  
  const baseURL = isDev() ? apiRootT : apiRootPr;

  const options = {
    baseURL: baseURL,
    headers: {
      Authorization: `Bearer ${authToken}`,
      Pragma: "no-cache"
    },
    timeout: 20000
  };

  const client = axios.create(options);

export default {
  get(url, conf = {}) {
    return getClient()
      .get(url, conf)
      .then(response => Promise.resolve(response))
      .catch(error => Promise.reject(error));
  },
}
`

**Comment by chatoo2412** [comment]
In my case, it happens when `baseURL` is set to path.

```js
const api = axios.create({ baseURL: '/api/' }); // problem
const api = axios.create({ baseURL: 'http://localhost:3000/api/' }); // OK

**Comment by mishawakerman** [comment]
Our workaround until #73 is merged:

**Before**
```
const api = axios.create({ baseURL: '/api/' });
```

**After**
```
const api = axios.create({ baseURL:  window.location.origin + '/api/' });
```

**Comment by maximn** [comment]
Any ETA for when is it going to be merged?

**Comment by mawrkus** [comment]
IMHO the "workaround" is so simple that I wouldn't try to patch axios-retry for this particular use case.

**Comment by downace** [comment]
Looks like it was fixed in [axios@0.19.1](https://github.com/axios/axios/releases/tag/0.19.1) (PR [#2391](https://github.com/axios/axios/pull/2391)). I've encountered the same issue, solved after updating axios.
